dmoj/judgeenv.py

```diff
@@ -98,12 +98,31 @@ def load_env(cli=False, testsuite=False):  # pragma: no cover
 
         dirs = env.problem_storage_root
         if dirs is not None:
+            get_path = lambda x, y: unicodify(os.path.normpath(os.path.join(x, y)))
             if isinstance(dirs, ConfigNode):
-                problem_dirs = tuple(unicodify(os.path.normpath(os.path.join(_root, dir))) for dir in dirs)
+                import re
+                rerec = re.compile(r'^(\d+)\s*:\s*(.*)$')
+                # Allow for recursion
+                def rec(dir, dep):
+                    if not dep: return [dir]
+                    ret = []
+                    for d in os.listdir(dir):
+                        nxt = os.path.join(dir, d)
+                        if os.path.isdir(nxt):
+                            ret += rec(nxt, dep - 1)
+                    return ret
+
+                problem_dirs = []
```

Comment:
Literally not how I said that issue should go. Arbitrary recursion is stupid since new directories won't be added.